rust/lance-index/src/vector/graph/storage.rs

```diff
@@ -36,4 +36,8 @@ pub trait VectorStorage: Send + Sync {
     /// Using dist calcualtor can be more efficient as it can pre-compute some
     /// values.
     fn dist_calculator(&self, query: &[f32]) -> Box<dyn DistCalculator>;
+
+    fn dist_calculator_from_id(&self, id: u32) -> Box<dyn DistCalculator>;
```

Comment:
I could refactor this into
```
enum Query {
    Vector(&[f32]),
    Node(u32),
}

fn dist_calculator(&self, query: Query) -> Box<dyn DistCalculator>;
```
if we think that's cleaner

Reply:
https://github.com/lancedb/lance/pull/2219
i'm changing it to `ArrayRef`